src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeFetchers.scala

```diff
@@ -291,4 +291,69 @@ trait BlocksDataFetchers {
     }
   }
 
+}
+
+/** Defines intances of `DataFetcher` for accounts-related data */
+trait AccountsDataFetchers {
+  //we require the cabability to log
+  self: LazyLogging =>
+  import cats.instances.future._
+  import cats.syntax.applicativeError._
+  import cats.syntax.applicative._
+  import JsonDecoders.Circe.decodeLiftingTo
+
+  implicit def fetchFutureContext: ExecutionContext
+
+  /* reduces repetion in error handling */
+  private def logWarnOnJsonDecoding[Encoded](message: String): PartialFunction[Throwable, Future[Unit]] = {
+    case decodingError: io.circe.Error =>
+      logger.warn(message, decodingError).pure[Future]
```

Comment:
circe errors are, in general, vague. Is the message here descriptive enough to compensate for that?

Reply:
This is one of the things we discussed the other day, I plan to spend some time parsing circe's errors to see if they can be translated to accurate and readable messages